### PR TITLE
Add batchnorm for conv, and the gluon version

### PR DESCRIPTION
**Pull request by fyears**
https://github.com/zackchase/mxnet-the-straight-dope/issues/151

I decide to add conv batch norm implementation.

### CODE REVIEW

**Comment by zackchase** [comment]
Thanks @fyears, can you please resolve conflicts before I can merge?

**Comment by fyears** [comment]
Hi @zackchase . After rebasing my branch, every thing should work fine now.
